kvbc/include/pruning_handler.hpp

```diff
@@ -189,4 +189,47 @@ class PruningHandler : public reconfiguration::IPruningHandler {
   mutable std::mutex pruning_status_lock_;
   mutable std::future<void> async_pruning_res_;
 };
+
+/*
+ * The read only pruning handler knows to answer only getLatestPruneableBlock.
+ * As it doesn't participant the consensus, the only thing we want to know is what is the latest reachable block.
+ */
+class ReadOnlyReplicaPruningHandler : public reconfiguration::IPruningHandler {
+ public:
+  ReadOnlyReplicaPruningHandler(IReader &ro_storage)
+      : ro_storage_{ro_storage},
+        signer_{bftEngine::ReplicaConfig::instance().replicaPrivateKey},
+        pruning_enabled_{bftEngine::ReplicaConfig::instance().pruningEnabled_},
+        replica_id_{bftEngine::ReplicaConfig::instance().replicaId} {}
+  bool handle(const concord::messages::LatestPrunableBlockRequest &,
```

Comment:
Now the reconfiguration engine needs to verify the `LatestPrunableBlock` message from the read-only replica too. We need to change the `PublicKeys()` function in the pruning handler so that it includes read-only replicas, correct?

Reply:
yes